### PR TITLE
Group good 'n' bad vulnerabilities and output them in tables

### LINKED ISSUE #29: [FEATURE] Output Vulnerabilities in a table
* **What are you trying to do?**

Similar to what we did on Nancy, output vulnerabilities in a table format.

Screen Shot 2020-05-19 at 5 20 32 PM

* **What feature or behavior is this required for?**

Not really required, but just a nice UX improvement. @ChurchRyan can answer questions if necessary!

* **How could we solve this issue? (Not knowing is okay!)**

On the branch `Summary` I have started a bit of this work, and used a table to output a Summary. If you follow that approach, you can replicate this for Vulnerabilities in `audit.py` in the `print_vulnerability` method. 

* **Anything else?**

HAVE A BLAST!!!!

cc @bhamail / @DarthHater

### PR DESCRIPTION
**Pull request by TheCodinator19**
Okie dokie, based on some best practices from user testing, now we have dependencies grouped by if they are vulnerable or not vulnerable.

Vulnerabilities output in a table, and a summary is shown at the end.
image

Fixes #29 

cc @bhamail / @DarthHater

### CODE REVIEW

**Comment by MichelKazi** [comment]
Currently in progress, looks like long strings in `terminaltables` objects don't like to text wrap so @TheCodinator19 and I will find a way to fix that and get some cleaner vulnerability tables :)

File: jake/audit/audit.py [hunk 1]
```diff
@@ -16,30 +16,59 @@
...
+    good = [x for x in results if len(x.get_vulnerabilities()) == 0]
+    bad = [x for x in results if len(x.get_vulnerabilities()) > 0]
+
+    if len(good) != 0:
+      print()
+      print("Non-Vulnerable Dependencies")
+      print()
+      for coordinate in good:
+        pkg_num += 1
+        total_vulns += self.print_result(coordinate, pkg_num, len(results))
```
Comment (bhamail) [inline jake/audit/audit.py:51]: I do not consider this a blocker, but now that we appear to split out "good" and "bad" results, the logic wherein `self.print_result()` feels a little overly complex and may no longer be needed? Or maybe I'm just misunderstanding how it works, and it is still needed to determine the true total if a result can have more than 1 vulnerability?

Either way, I would not consider adding any change related to the above comment a requirement for this PR, in fact, such changes would probably muddy the water.

File: jake/audit/audit.py [hunk 2]
```diff
@@ -16,30 +16,59 @@
...
-    for coordinate in results:
-      pkg_num += 1
-      total_vulns += self.print_result(coordinate, pkg_num, len(results))
+    table_instance = DoubleTable(table_data, "Summary")
+
+    print()
+
+    print(table_instance.table)
 
     return total_vulns
```
Comment (bhamail) [inline jake/audit/audit.py:73]: Is it silly that the little unit test: `test/test_audit.py->test_call_audit_results_prints_output()`  that verifies the 3 return count makes me so happy? ;)

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,28 @@
...
+*~
+
+# Auto-generated tag files
+tags
+
+# Persistent undo
+[._]*.un~
+
+# Coc configuration directory
+.vim
```
[the remark this answers is not in the document]
Comment (DarthHater) [reply .gitignore:24]: @MichelKazi is a 40 year old in a 22 year olds body

File: jake/audit/audit.py [hunk 4]
```diff
@@ -16,30 +16,59 @@
...
 from typing import List
+from textwrap import wrap
 from colorama import Fore
+from terminaltables import DoubleTable
 
 from ..types.coordinateresults import CoordinateResults
 from ..types.vulnerabilities import Vulnerabilities
 
-class Audit():
+class Audit:
```
[the remark this answers is not in the document]
Comment (DarthHater) [reply jake/audit/audit.py:26]: You don't really need to do the `()`, basically, `class Audit` is sufficient.

File: jake/audit/audit.py [hunk 5]
```diff
@@ -16,30 +16,59 @@
...
   """ Audit does the business, it prints results from OSS Index to the standard out """
+
   def __init__(self, quiet=False):
-    self._log = logging.getLogger('jake')
+    self._log = logging.getLogger("jake")
```
[the remark this answers is not in the document]
Comment (DarthHater) [reply jake/audit/audit.py:30]: Basically the same in the `python` world, but the added benefit of double quotes is you can have a single quote in the string?

File: jake/audit/audit.py (hunk 1 above)
Comment (DarthHater) [reply jake/audit/audit.py:51]: Yeah this change was a step in the direction you are talking about. I figured I wouldn't bite it off in this PR, but it seems to make sense to do exactly what you are saying!

File: jake/audit/audit.py [hunk 6]
```diff
@@ -51,20 +80,19 @@ def print_result(self, coordinate: CoordinateResults, number, length):
     """
     if len(coordinate.get_vulnerabilities()) == 0:
       if not self._quiet:
-        self.do_print("[{}/{}] - {} - no known vulnerabilities for this version"
-                      .format(
-                          number,
-                          length,
-                          coordinate.get_coordinates()), 0)
+        self.do_print(
+            f"[{number}/{length}] - {coordinate.get_coordinates()}",
```
[the remark this answers is not in the document]
Comment (DarthHater) [reply jake/audit/audit.py:84]: I bet it's your space count? Maybe?

**Pull request merged**